olive/common/hf/mlflow.py

```diff
@@ -37,14 +40,23 @@ def get_pretrained_name_or_path(model_name_or_path: str, name: str) -> str:
 
     parent_dir = Path(model_name_or_path).resolve()
 
-    # assumed to be an mlflow model
-    pretrained_path = parent_dir / "data" / name
-    if pretrained_path.exists():
-        return str(pretrained_path)
+    mlflow_data = load_mlflow_model_data(model_name_or_path)
+    flavors = mlflow_data.get("flavors", {}).keys()
+
+    if "hftransformersv2" in flavors:
+        path = parent_dir / "data" / name
+        if path.exists():
+            return str(path)
+
```

Comment:
the logic from line 45-48 is still needed here. There are some models where the data subdirectories are not present

Reply:
Line 46 is `parent_dir / "data" / "model"`, do you mean `parent_dir / "model"`?

Reply:
`parent_dir / "data" / "model"` An example model https://ml.azure.com/models/Llama-2-7b-chat/version/24/catalog/registry/azureml-meta

I don't remember the details, but I think it depends on whether some experimental features are turned on when the model was saved.

Reply:
btw, could you add these models as comments for each case so we have context in the future? Thanks!

Reply:
Done